### DISCUSSION TITLE
Mouse HID Descriptor: buttons + padding => more buttons

### DISCUSSION THREAD

**Discussion by jalmeroth** [hrvach/deskhop#176] [Ideas]
Hi,

while we currently modify the mouse descriptors, I was thinking of extending the `HID_USAGE_PAGE_BUTTON` from 5x 1 bit (for Left, Right, Middle, Backward, Forward buttons) plus 3x 1 bit of padding to actually full 8x 1 bit.

To my understanding this would not change the resulting HID reports for users with five or less buttons, but make the descriptor slightly smaller and would enable users with mice that have up to 8 buttons.

now:
![swappy-20241122_191110](https://github.com/user-attachments/assets/37f8e0f6-a723-45f2-980a-51313a6489df)

without padding:
![swappy-20241122_191111](https://github.com/user-attachments/assets/aac8117c-ed8b-468f-9f60-9e3b243dc836)

What do you think?

**Comment by hrvach** [comment]
I think it's a great idea, I'm including it in the next release.
